### ISSUE TITLE
Spectre V2: add check for cpuinfo "spec_ctrl" ?

### ISSUE THREAD

**Issue by Feandil** [speed47/spectre-meltdown-checker#36]
I've seen a weird case on a VM:
- Reading from the msr works (the CPU has an updated microcode)
- The kernel has been compiled with ibrs support
- The kernel choose to disable ibrs support
- /proc/cpuinfo does not contain spec_ctrl ibpb_support

I believe this to be a bad qemu package/runtime/configuration, but I was wondering: would it make sense to add a check for the spec_ctrl flag in /proc/cpuinfo?

**Comment by speed47** [comment]
Things are moving really fast on the LKML and they're changing the code several times per day, including the "flags" that would be visible from /proc or /sys.
Do you have some kernel source code or diff/patch showing a `spec_ctrl` or `ipbp_support` flag exported to `/proc/cpuinfo`?

**Comment by Feandil** [comment]
It must be in the RedHat kernel sources, but these are only released as source code and hard to parse...
Latest (right now) sources can be downloaded from https://git.centos.org/sources/kernel/c7/b0cf50a45ee77f54c233f0f8d8f57324e266f3d8 (theoretical file name: linux-3.10.0-693.11.6.el7.tar.xz).

If you think it's not worth it and not generic, let's ignore it ;)

**Comment by speed47** [comment]
From your archive, I found `Documentation/spec_ctrl.txt` which explains a bit, and references the `/sys/kernel/debug/x86/ibrs_enabled` file that we've included in the script.
So I think looking for this file should be sufficient, and the script already does it ;)

Closing the issue to tidy up a bit, feel free to reopen if needed
